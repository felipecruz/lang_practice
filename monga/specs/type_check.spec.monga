void ret_void ()
{
    return;
}

int ret_int (int x)
{
    int a;
    int[] w;
    int[] q;

    x = 3;
    q = w;

    w = new int[10];
    a = ret_int (1);

    a = ret_int (ret_int (w[0]));

    x = w[1];
    q[0] = w[1];
    q[1] = x;

    a = 1 + 2;
    a = a * 10;
    a = x / a;

    if (a == 10) {}
    if ((x + a) == -10) {}
    if (x >= 10) {}
    if (a <= 10) {}
    if (a > 10) {}
    if (a < 10 || (x * 2 + (a / 10))) {}

    if (!1) {

    }

    while (-1) {

    }

    if (-a) {}
    else {}

    if (!x) {}

    if (a && x) {}
    if (a && 1) {}
    if (0 && x) {}

    ret_float(3.0);

    return x;
}

float ret_float (float f)
{
    float w;
    w = 1 + 1.0;
    w = 1.0 + 1 + 2.0;
    w = 1;
    f = ret_float (2.0);
    f = ret_float (1.0);

    if (1 > 0) {

    }

    return f;
}

int f_all (int x, float f, char[] st)
{
    return f_all (x, f, st);
}

char[] ret_char_array ()
{
    char[] st;
    char[] a;
    char q;
    int x;
    q = 10;
    x = 10;
    q = x;
    st = "uma string";
    st = a;
    ret_int (1);
    x = f_all (1, 2.0, "aa");
    return "abcd";
}

int main (int argc, char[] argv)
{
    int x;
    x = 1;
    return ret_int (x + 2);

}
