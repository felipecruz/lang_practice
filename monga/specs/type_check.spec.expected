Checking Func int_ret_char
Checking return type.. 
Get Expression Var Type...
Checking Func char_ret_int
Checking return type.. 
Get Expression Var Type...
Checking Func ret_void
Checking return type.. 
Checking Func ret_int
Checking assignment types... 
Checking assignment types... 
Get Expression Var Type...
Checking assignment types... 
Checking assignment types... 
Get call return type..
Checking assignment types... 
Get call return type..
Get call return type..
Checking assignment types... 
Get Expression Var Type...
Checking assignment types... 
Get Expression Var Type...
Checking assignment types... 
Get Expression Var Type...
Checking assignment types... 
Checking assignment types... 
Checking assignment types... 
Get Expression Var Type...
Checking assignment types... 
Get Expression Var Type...
Get Expression Var Type...
Get Expression Var Type...
Get Expression Var Type...
Get Expression Var Type...
Get Expression Var Type...
Get Expression Var Type...
Get Expression Var Type...
Get Expression Var Type...
Get Expression Var Type...
Get Expression Var Type...
Get Expression Var Type...
Get Expression Var Type...
Get Expression Var Type...
Get Expression Var Type...
Get Expression Var Type...
Get Expression Var Type...
Checking return type.. 
Get Expression Var Type...
Checking Func ret_float
Checking assignment types... 
Int Change Exp to Exp with cast
Checking assignment types... 
Checking assignment types... 
Checking assignment types... 
Checking assignment types... 
Checking assignment types... 
Get call return type..
Checking assignment types... 
Get call return type..
Checking return type.. 
Get call return type..
Checking Func f_all
Checking return type.. 
Get call return type..
Checking Func ret_char_array
Checking assignment types... 
Checking assignment types... 
Checking assignment types... 
Get Expression Var Type...
Checking assignment types... 
Checking assignment types... 
Get Expression Var Type...
Checking assignment types... 
Get call return type..
Checking return type.. 
Checking Func main
Checking assignment types... 
Checking return type.. 
Get call return type..
